src/session.coffee

```diff
@@ -165,20 +170,43 @@ define [
 
       full_options = expires: Settings.cookies.full.session.duration
       full_options.domain = @domain if @domain
-      Biskoto.set Settings.cookies.full.session.name, session_data, full_options if @cookie_policy == 'full'
+      if @cookie_policy == 'full' && @shop_cookie_policy == 'full'
+        Biskoto.set Settings.cookies.full.session.name, session_data, full_options
 
       # Delete full cookie if user changed preferences
       full_cookie = Biskoto.get Settings.cookies.full.session.name
-      Biskoto.expire Settings.cookies.full.session.name, full_options if full_cookie && @cookie_policy == 'basic'
+      if full_cookie && (@cookie_policy == 'basic' || @shop_cookie_policy == 'basic')
+        Biskoto.expire Settings.cookies.full.session.name, full_options
 
     _createMetadataCookie: ->
       options = { domain: @domain } if @domain
 
       Biskoto.set Settings.cookies.basic.meta.name, @metadata, options
-      Biskoto.set Settings.cookies.full.meta.name, @metadata, options if @cookie_policy == 'full'
+      if @cookie_policy == 'full' && @shop_cookie_policy == 'full'
+        Biskoto.set Settings.cookies.full.meta.name, @metadata, options
 
       # Delete full cookie if user changed preferences
       full_cookie = Biskoto.get Settings.cookies.full.meta.name
-      Biskoto.expire Settings.cookies.full.meta.name, options if full_cookie && @cookie_policy == 'basic'
+      if full_cookie && (@cookie_policy == 'basic' || @shop_cookie_policy == 'basic')
+        Biskoto.expire Settings.cookies.full.meta.name, options
+
+    _unsetFullCookiesBasedOnShopCookiePolicy: ->
+      if @analytics_session
+        # Delete full cookie if user changed preferences
+        full_options = expires: Settings.cookies.full.session.duration
+        full_options.domain = @domain if @domain
+        full_session_cookie = Biskoto.get Settings.cookies.full.session.name
+        Biskoto.expire Settings.cookies.full.session.name, full_options if full_session_cookie
+
+      if @metadata
+        options = { domain: @domain } if @domain
+
+        # Delete full cookie if user changed preferences
+        full_metadata_cookie = Biskoto.get Settings.cookies.full.meta.name
+        Biskoto.expire Settings.cookies.full.meta.name, options if full_metadata_cookie
+
+      full_cache_options = expires: Settings.cookies.full.analytics.duration
+      full_cache_options.domain = @domain if @domain
+      Biskoto.expire Settings.cookies.full.analytics.name, full_cache_options
```

Comment:
Is there a reason why we don't check here if a `Biskoto.get(Settings.cookies.full.analytics.name)` exists, as we do in the previous cookies?